src/category_theory/instances/Top/presheaf_of_functions.lean

```diff
@@ -0,0 +1,84 @@
+import category_theory.instances.Top.presheaf
+import category_theory.instances.TopCommRing.basic
+import category_theory.yoneda
+import group_theory.submonoid
+import ring_theory.subring
+import topology.algebra.continuous_functions
+
+universes v u
+
+open category_theory
+open category_theory.instances
+open topological_space
+
+open category_theory
+open category_theory.instances
+open topological_space
+
+namespace category_theory.instances.Top
+
+variables (X : Top.{v})
+
+def presheaf_to_Top (T : Top.{v}) : X.presheaf (Type v) :=
+(opens.to_Top X).op ⋙ (yoneda.obj T)
+
+-- TODO upgrade the result to TopCommRing?
+def continuous_functions (X : Top.{v}ᵒᵖ) (R : TopCommRing.{v}) : CommRing.{v} :=
+{ α := (unop X) ⟶ (TopCommRing.forget_to_Top.obj R),
+  str := _root_.continuous_comm_ring } -- but infer_instance doesn't work?
```

Comment:
The reason `infer_instance` doesn't work is that the type of `continuous_comm_ring` is `comm_ring { f : α → β | continuous f }`, but the type you're trying to synthesize is `(unop X) ⟶ (TopCommRing.forget_to_Top.obj R)`. These are definitionally equal, but only after unfolding the `category` instance for `Top`. Type-class inference will not unfold definitions, unless they are marked as `[reducible]` (or `[inline]`(?)).

Reply:
Ah, of course. Thanks.